integration/helpers.go

```diff
@@ -437,3 +438,34 @@ func generateDocuments(startID, endID int32) bson.A {
 
 	return docs
 }
+
+// CreateNestedDocument creates a mock nested Bson document. The nesting level is based on integer parameter.
+func CreateNestedDocument(n int) bson.M {
```

Comment:
It seems to me that we test for nested documents but don't test documents nested in arrays, nested in documents, ...; Can we modify this function to produce this type of nesting?